src/attributes.md

```diff
@@ -582,52 +172,133 @@ pub fn f() {}
 
 > Note: `rustc` currently recognizes the tools "clippy" and "rustfmt".
 
+## Built-in attributes
+
+The following is an index of all built-in attributes.
+
+- Conditional Compilation
```

Comment:
We should follow the style of headings on the page, and lowercase the non-first word.